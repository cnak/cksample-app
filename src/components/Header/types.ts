import { NavigationProps } from '../../navigation/types';

export interface HeaderTextProps {
    headerText: string;
}
export interface HeaderProps extends HeaderTextProps, NavigationProps {}
