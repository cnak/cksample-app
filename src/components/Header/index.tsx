import { MaterialIcons } from '@expo/vector-icons';
import React from 'react';
import { Platform, StatusBar } from 'react-native';
import { getStatusBarHeight } from 'react-native-status-bar-height';
import styled, { css } from 'styled-components/native';

import StyledText from '../StyledText';
import { BRAND_DARK, WHITE } from '../colors';

import { HeaderProps } from './types';

const isIOS = Platform.OS === 'ios';

const StatusBarSeparator = styled.View<{ height: number }>`
    height: ${({ height }) => height}px;
`;

const VERTICAL_PADDING = 15;
const HORIZONTAL_PADDING = 15;
const ICON_SIZE = 30;

const Header = styled.View<{ canGoBack?: boolean }>`
    background-color: ${BRAND_DARK};
    height: ${ICON_SIZE + 2 * VERTICAL_PADDING}px;
    padding: ${VERTICAL_PADDING}px 0;
    flex-direction: row;
    justify-content: ${({ canGoBack }) => (canGoBack ? 'flex-start' : 'center')};
    align-items: center;
`;

const buttonPaddingCss = css`
    padding: ${isIOS ? 0 : VERTICAL_PADDING}px ${HORIZONTAL_PADDING}px;
`;

const BackButton = styled.TouchableOpacity`
    ${buttonPaddingCss};
`;

const HeaderText = styled(StyledText)`
    color: ${WHITE};
    font-size: 26px;
    font-weight: 300;
    padding: ${0}px 0;
    margin-left: ${-HORIZONTAL_PADDING}px;
`;

const HeaderWithStatusBar = ({ canGoBack, goBack, headerText }: HeaderProps) => {
    const onBackButtonPress = canGoBack && goBack ? () => goBack() : () => {};
    return (
        <>
            {isIOS && <StatusBar barStyle="default" />}

            <StatusBarSeparator height={getStatusBarHeight()} />

            <Header canGoBack={canGoBack}>
                {canGoBack && (
                    <BackButton
                        onPress={onBackButtonPress}
                        accessibilityRole="button"
                        accessibilityLabel="Go back"
                        testID="BackButton"
                    >
                        <MaterialIcons color={WHITE} name="arrow-back" size={ICON_SIZE} />
                    </BackButton>
                )}

                <HeaderText style={{ marginLeft: canGoBack ? 32 : 0 }} accessibilityLabel="Header">
                    {headerText}
                </HeaderText>
            </Header>
        </>
    );
};

export default HeaderWithStatusBar;
