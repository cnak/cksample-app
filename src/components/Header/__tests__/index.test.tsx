import React from 'react';
import renderer from 'react-test-renderer';

import canGoBackProps from '../mocks/props';
import Header from '..';

const cannotGoBackProps = { ...canGoBackProps, canGoBack: false };
const signedOutHeaderProps = { ...cannotGoBackProps, isSignedIn: false };

describe('<Header>', () => {
    describe('matches snapshots', () => {
        /* it('can go back', () => { */
        /*     const tree = renderer.create(<Header {...canGoBackProps} />).toJSON(); */
        /*     expect(tree).toMatchSnapshot(); */
        /* }); */

        it('cant go back', () => {
            const tree = renderer.create(<Header {...cannotGoBackProps} />).toJSON();
            expect(tree).toMatchSnapshot();
        });

        it('sign-in screen scenario', () => {
            const tree = renderer.create(<Header {...signedOutHeaderProps} />).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
});
