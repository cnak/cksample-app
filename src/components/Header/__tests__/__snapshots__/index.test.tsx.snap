// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header> matches snapshots cant go back 1`] = `
Array [
  <View
    height={20}
    style={
      Array [
        Object {
          "height": 20,
        },
      ]
    }
  />,
  <View
    canGoBack={false}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#0B0C0C",
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "center",
          "paddingBottom": 15,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 15,
        },
      ]
    }
  >
    <Text
      accessibilityLabel="Header"
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Helvetica",
            "fontSize": 18,
            "fontWeight": "bold",
            "marginLeft": -15,
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 0,
          },
          Object {
            "marginLeft": 0,
          },
        ]
      }
    >
      Header example
    </Text>
  </View>,
]
`;

exports[`<Header> matches snapshots sign-in screen scenario 1`] = `
Array [
  <View
    height={20}
    style={
      Array [
        Object {
          "height": 20,
        },
      ]
    }
  />,
  <View
    canGoBack={false}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#0B0C0C",
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "center",
          "paddingBottom": 15,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 15,
        },
      ]
    }
  >
    <Text
      accessibilityLabel="Header"
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Helvetica",
            "fontSize": 18,
            "fontWeight": "bold",
            "marginLeft": -15,
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 0,
          },
          Object {
            "marginLeft": 0,
          },
        ]
      }
    >
      Header example
    </Text>
  </View>,
]
`;
